tests/test_utils.py

```diff
@@ -243,3 +265,31 @@ def test_check_status_code_for_deprecated_pypi_url(repo_url):
     # value of Verbose doesn't matter for this check
     with pytest.raises(exceptions.UploadToDeprecatedPyPIDetected):
         utils.check_status_code(response, False)
+
+
+@pytest.mark.parametrize(
+    "repo_url", ["https://pypi.python.org", "https://testpypi.python.org"],
+)
+def test_check_status_code_for_missing_status_code(capsys, repo_url):
+    """
+    Test if the status code returned is not an explicitly checked one
+    """
+    response = pretend.stub(
+        status_code=403,
+        url=repo_url,
+        raise_for_status=pretend.raiser(requests.exceptions.HTTPError),
+        text="Forbidden",
+    )
+
+    with pytest.raises(requests.exceptions.HTTPError):
```

Comment:
In the future, this can just be `requests.HTTPError`.

Reply:
Thanks, should I fix this in another PR ?

Reply:
If you like, though I'd be fine with it being part of #596.

Reply:
Thanks, committed with https://github.com/pypa/twine/pull/596/commits/d926a3fa7d89b04e40a1a83a62b2b363a2c8523f